starfish/pipeline/features/spots/detector/detect.py

```diff
@@ -0,0 +1,222 @@
+    image : Union[np.ndarray, xr.DataArray],
+        3-d volume in which to measure intensities
+    spots : pd.DataFrame
+        SpotAttributes table containing coordinates and radii of spots
+    measurement_function : Callable[[Sequence], Number]) \
+        Function to apply over the spot volumes to identify the intensity (e.g. max, mean, ...)
+
+    Returns
+    -------
+    pd.Series :
+        Intensities for each spot in SpotAttributes
+
+    """
+
+    def fn(row: pd.Series) -> Number:
+        data = image[
+            row['z_min']:row['z_max'],
+            row['y_min']:row['y_max'],
+            row['x_min']:row['x_max']
+        ]
+        return measurement_function(data)
+
+    # construct an inclusive bounding box for each spot before it is submitted to measure
+    inclusive_radius = (np.ceil(spots[Features.SPOT_RADIUS]) + 1).astype(int)
+    for v, max_size in zip(['z', 'y', 'x'], image.shape):
+        spots[f'{v}_min'] = np.clip(spots[v] - inclusive_radius, 0, None)
+        spots[f'{v}_max'] = np.clip(spots[v] + inclusive_radius, None, max_size)
+
+    return spots[['z_min', 'z_max', 'y_min', 'y_max', 'x_min', 'x_max']].astype(int).apply(
+        fn,
+        axis=1
+    )
+
+
+def measure_spot_intensities(
+        data_image: ImageStack,
+        spot_attributes: pd.DataFrame,
+        measurement_function: Callable[[Sequence], Number], ) \
+        -> IntensityTable:
+    """given spots found from a reference image, find those spots across a data_image
+
+    Parameters
+    ----------
+    data_image : ImageStack
+        ImageStack containing multiple volumes for which spots' intensities must be calculated
+    spot_attributes : pd.Dataframe
+        Locations and radii of spots
+    measurement_function : Callable[[Sequence], Number]) \
+        Function to apply over the spot volumes to identify the intensity (e.g. max, mean, ...)
+
+    Returns
+    -------
+    IntensityTable :
+        3d tensor of (spot, channel, round) information for each coded spot
+
+    """
+
+    # determine the shape of the intensity table
+    n_ch = data_image.shape[Indices.CH]
+    n_round = data_image.shape[Indices.ROUND]
+    spot_attribute_index = dataframe_to_multiindex(spot_attributes)
+    image_shape: Tuple[int, int, int] = data_image.raw_shape[2:]
+
+    # construct the empty intensity table
+    intensity_table = IntensityTable.empty_intensity_table(
+        spot_attributes=spot_attribute_index,
+        n_ch=n_ch,
+        n_round=n_round,
+        image_shape=image_shape
+    )
+
+    # fill the intensity table
+    indices = product(range(n_ch), range(n_round))
+    for c, r in indices:
+        image, _ = data_image.get_slice({Indices.CH: c, Indices.ROUND: r})
+        blob_intensities: pd.Series = measure_spot_intensity(
+            image, spot_attributes, measurement_function)
+        intensity_table[:, c, r] = blob_intensities
+
+    return intensity_table
+
+
+def concatenate_spot_attributes_to_intensities(
```

Comment:
Previously there was no logic to stick together multiple SpotAttributes tables into a single IntensityTable. Now that we implement the LocalMaxPeakFinder approach, we needed one. This serves that role.